src/main/java/com/aliyun/lindorm/tsdb/client/impl/LindormTSDBClientImpl.java

```diff
@@ -222,6 +223,24 @@ public CompletableFuture<WriteResult> write(String database, List<Record> record
         return this.batchProcessor.append(database, records, nowMs);
     }
 
+    @Override
+    public CompletableFuture<WriteResult> write(String database, Record record, List<String> clusterIdList) {
+        if (clusterIdList == null) {
+            return write(database, record);
+        }
+        long nowMs = System.currentTimeMillis();
+        return this.batchProcessor.append(database, record, nowMs, clusterIdList);
+    }
+
+    @Override
+    public CompletableFuture<WriteResult> write(String database, List<Record> records, List<String> clusterIdList) {
+        if (clusterIdList == null) {
+            return write(database, records);
+        }
+        long nowMs = System.currentTimeMillis();
```

Comment:
if (records == null || records.isEmpty()) {
            throw new IllegalArgumentException("The records must be not null or empty.");
        }